registry/registry.go

```diff
@@ -0,0 +1,62 @@
+package registry
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"golang.org/x/net/context"
+
+	"github.com/docker/cli/config/configfile"
+	"github.com/docker/docker/api/types"
+	registrytypes "github.com/docker/docker/api/types/registry"
+	"github.com/docker/docker/client"
+	dockerregistry "github.com/docker/docker/registry"
+)
+
+// ElectAuthServer returns the default registry to use (by asking the daemon)
+func ElectAuthServer(ctx context.Context, c client.APIClient) (string, []error, error) {
```

Comment:
I noticed you added an `error` return value, but it seems to always be `nil`. Can we remove it?